### DISCUSSION TITLE
Upgrading to thymeleaf-spring6-3.1.2.RELEASE causing errors.

### DISCUSSION THREAD

**Discussion by jrojas-efx** [thymeleaf/thymeleaf#971] [General]
Hi, i'm trying to upgrade thymeleaf-spring6 to the latest version but i'm getting this error in several projects:

java.lang.NoSuchMethodError: 'java.lang.String org.thymeleaf.util.ExpressionUtils.normalize(java.lang.String)'

does someone managed to upgrade without problems?

**Comment by danielfernandez** [comment]
That method was added in 3.1.2. Could you please verify that you are adding to the classpath of your application both`thymeleaf-3.1.2.RELEASE.jar` and `thymeleaf-spring6-3.1.2.RELEASE.jar`? That exception looks as if the "core" jar was still in a previous version…

**Reply by jrojas-efx** [reply]
You are right Daniel, I added `thymeleaf-3.1.2.RELEASE.jar` to the classpath and it worked fine. Thank you very much
